backend/server.js

```diff
@@ -961,10 +955,37 @@ app.get("/qInstances/:qiid/:filename", function(req, res) {
                 if (!_(info).has("clientFiles")) {
                     return sendError(res, 500, "Question does not have clientFiles, qid: " + fileInfo.qid);
                 }
-                if (!_(info.clientFiles).contains(fileInfo.filename)) {
-                    return sendError(res, 404, "Access denied to '" + fileInfo.filename + "' for qid: " + fileInfo.qid);
+
+                if (_(info).has("clientTemplates") && _(info.clientTemplates).contains(fileInfo.filename)) {
+                    // File is template which needs to be rendered
+                    fs.readFile(path.join(fileInfo.root, fileInfo.filePath), function(err, data) {
```

Comment:
I think express and/or node and/or apache already prevents this. At least, `/clientFiles/../courseInfo.json` doesn't work, and there's nothing in `server.js` to prevent it.

... actually, some further testing against a local PL server indicates that `sendFile` by default doesn't trust relative paths. However, I'm pretty sure something else on the prod server (likely apache) is resolving the request path to `/courseInfo.json`